components/rhs_header_post/index.ts

```diff
@@ -33,15 +34,21 @@ import {matchUserMentionTriggersWithMessageMentions} from '../../utils/post_util
 
 import RhsHeaderPost from './rhs_header_post';
 
+type UserThreadOrSythetic = UserThread | UserThreadSynthetic;
 type OwnProps = Pick<ComponentProps<typeof RhsHeaderPost>, 'rootPostId'>
 
 function mapStateToProps(state: GlobalState, {rootPostId}: OwnProps) {
     const root = getPost(state, rootPostId);
     const currentUserMentionKeys = getCurrentUserMentionKeys(state);
-    const rootMessageMentionKeys = allAtMentions(root.message);
-    const thread = getThreadOrSynthetic(state, root);
-    const isMentionedInRootPost = thread.reply_count === 0 &&
-        matchUserMentionTriggersWithMessageMentions(currentUserMentionKeys, rootMessageMentionKeys);
+
+    let isMentionedInRootPost = false;
+    let thread = {} as UserThreadOrSythetic;
```

Comment:
Instead of making a fake object here, could we perhaps make it so that `getThreadOrSynthetic` can handle the case where `root` is undefined? This could be making the `RhsRootPost` render constantly by passing a new object on each state change, and I'd be worried about other things failing because the fields of this empty object are undefined.

Reply:
Changing `getThreadOrSynthetic` would need to change every use to handle the `null` case,
so I'll refactor a bit to handle the new object.

Reply:
BTW, the root cause of this is that `getSelectedPost` can return a fake post which is desired.
https://github.com/mattermost/mattermost-webapp/blob/a468fdb80e61c876462ed56f133c562fc95db737/selectors/rhs.ts#L64-L85

Reply:
I meant that we could use the fake thread returned by `getThreadOrSynthetic` instead of making a new empty object here by adding an extra null check. I'm not the biggest fan of fake objects, but at least the one returned by `getThreadOrSynthetic` has all the fields provided so that we won't accidentally cause errors in code using the object.

```
export function getThreadOrSynthetic(state, rootPost) {
    const thread = getThreads(state)[rootPost?.id]; // Adding ?. here lets rootPost be undefined

    if (thread?.id) {
        return thread;
    }

    return {/* something else fake but with all the required fields*/};
}
```